compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1299,6 +1299,10 @@ void handleDeclaredVarInForeach(BLangVariable variable, BType rhsType, SymbolEnv
                 recursivelySetFinalFlag(tupleVariable);
                 break;
             case RECORD_VARIABLE:
+                if (rhsType.tag == TypeTags.ANYDATA || rhsType.tag == TypeTags.ANY) {
+                    dlog.error(variable.pos, DiagnosticErrorCode.INVALID_RECORD_BINDING_PATTERN, rhsType);
+                    return;
+                }
```

Comment:
Should we fail this at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolEnter#validateRecordVariable instead?

By removing the any and anydata cases at https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java#L2266?

Reply:
I thought of doing it, but there are multiple usages of  validateRecordVariables. Basically it is used for validating simple assignments to record variables as well

Reply:
Can you please share an example? Wondering if those should actually result in compilation errors too?

Reply:
@MaryamZi I actually double checked it, We can of course move it to SymbolEnter. One of the negative test was failing with the current changes and that test is  kind of not correct. I will send the changes